api/debug/flags.go

```diff
@@ -136,37 +177,71 @@ func CreateLogDir(logDir string) {
+		ostream = log.StreamHandler(output, log.JsonFormat())
+	} else {
+		usecolor := (isatty.IsTerminal(os.Stderr.Fd()) && os.Getenv("TERM") != "dumb")
+		if usecolor {
+			output = colorable.NewColorableStderr()
+		}
+		ostream = log.StreamHandler(output, log.TerminalFormat(usecolor))
+	}
+	glogger.SetHandler(ostream)
+
 	// logging
-	log.PrintOrigins(ctx.GlobalBool(debugFlag.Name))
+	log.PrintOrigins(ctx.GlobalBool(getFlagName(ctx, debugFlag, legacyDebugFlag, "(v1.12.0)")))
 	log.ChangeGlobalLogLevel(glogger, log.Lvl(ctx.GlobalInt(verbosityFlag.Name)))
 	glogger.Vmodule(ctx.GlobalString(vmoduleFlag.Name))
-	glogger.BacktraceAt(ctx.GlobalString(backtraceAtFlag.Name))
+
+	if location := ctx.GlobalString(getFlagName(ctx, backtraceAtFlag, legacyBacktraceAtFlag, "(v1.12.0)")); location != "" {
+		glogger.BacktraceAt(location)
+	}
 	log.Root().SetHandler(glogger)
 
 	// profiling, tracing
-	runtime.MemProfileRate = ctx.GlobalInt(memprofilerateFlag.Name)
-	Handler.SetBlockProfileRate(ctx.GlobalInt(blockprofilerateFlag.Name))
+	if memprofilerateFlagName := getFlagName(ctx, memprofilerateFlag, legacyMemprofilerateFlag, "(v1.12.0)"); memprofilerateFlagName != "" {
+		runtime.MemProfileRate = ctx.GlobalInt(memprofilerateFlagName)
+	}
+
+	if blockprofilerateFlagName := getFlagName(ctx, blockprofilerateFlag, legacyBlockprofilerateFlag, "(v1.12.0)"); blockprofilerateFlagName != "" {
+		Handler.SetBlockProfileRate(ctx.GlobalInt(blockprofilerateFlagName))
+	}
+
 	if traceFile := ctx.GlobalString(traceFlag.Name); traceFile != "" {
 		if err := Handler.StartGoTrace(traceFile); err != nil {
 			return err
 		}
 	}
-	if cpuFile := ctx.GlobalString(cpuprofileFlag.Name); cpuFile != "" {
-		if err := Handler.StartCPUProfile(cpuFile); err != nil {
+	if cpuProfile := ctx.GlobalString(getFlagName(ctx, cpuprofileFlag, legacyCpuprofileFlag, "(v1.12.0)")); cpuProfile != "" {
+		if err := Handler.StartCPUProfile(cpuProfile); err != nil {
 			return err
 		}
 	}
-	Handler.memFile = ctx.GlobalString(memprofileFlag.Name)
+
+	Handler.memFile = ctx.GlobalString(getFlagName(ctx, memprofileFlag, legacyMemprofileFlag, "(v1.12.0)"))
 
 	// pprof server
 	if ctx.GlobalBool(pprofFlag.Name) {
-		addr := ctx.GlobalString(pprofAddrFlag.Name)
-		port := ctx.GlobalInt(pprofPortFlag.Name)
-		Handler.StartPProf(&addr, &port)
+		addr := ctx.GlobalString(getFlagName(ctx, pprofAddrFlag, legacyPprofAddrFlag, "(v1.12.0)"))
+		pprofPort := ctx.GlobalInt(getFlagName(ctx, pprofPortFlag, legacyPprofPortFlag, "(v1.12.0)"))
+		Handler.StartPProf(&addr, &pprofPort)
 	}
 	return nil
 }
 
+func getFlagName(ctx *cli.Context, flag cli.Flag, legacyFlag cli.Flag, deprecateVersion string) string {
+	if ctx.GlobalIsSet(flag.GetName()) {
+		return flag.GetName()
+	}
+	if ctx.GlobalIsSet(legacyFlag.GetName()) {
+		logger.Warn("The flag " + legacyFlag.GetName() + " is deprecated and will be removed in the future " + deprecateVersion + ", please use " + flag.GetName())
+		return legacyFlag.GetName()
+	}
+	return ""
```

Comment:
```suggestion
	logger.Warn("The flag " + legacyFlag.GetName() + " is deprecated and will be removed in the future " + deprecateVersion + ", please use " + flag.GetName())
	return legacyFlag.GetName()
```
How about removing `return ""`? If `legacyFlag` is not set, this will return `""`.